pkg/sql/plan2/explain/build_node_explain.go

```diff
@@ -0,0 +1,382 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package explain
+
+import "github.com/matrixorigin/matrixone/pkg/pb/plan"
+
+var _ NodeDescribe = &NodeDescribeImpl{}
+
+type NodeDescribeImpl struct {
+	Node *plan.Node
+}
+
+func NewNodeDescriptionImpl(node *plan.Node) *NodeDescribeImpl {
+	return &NodeDescribeImpl{
+		Node: node,
+	}
+}
+
+func (ndesc NodeDescribeImpl) GetNodeBasicInfo(options *ExplainOptions) string {
```

Comment:
pointer?

Reply:
This method should use the pointer as the receiver instead